src/devices/CharacterLcd/LcdConsole.cs

```diff
@@ -0,0 +1,585 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Drawing;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Globalization;
+
+namespace Iot.Device.CharacterLcd
+{
+    /// <summary>
+    /// This is a high-level interface to an LCD display.
+    /// It supports automatic wrapping of text, automatic scrolling and code page mappings.
+    /// This class is thread safe, however using Write from different threads may lead to unexpected results, since the order is not guaranteed.
+    /// </summary>
+    public sealed class LcdConsole : IDisposable
+    {
+        private readonly object _lock;
+        private readonly bool _shouldDispose;
+        private ICharacterLcd _lcd;
+
+        /// <summary>
+        /// The text currently on the display (required for arbitrary scrolling)
+        /// </summary>
+        private StringBuilder[] _currentData;
+        private LineWrapMode _lineFeedMode;
+        private TimeSpan _scrollUpDelay;
+        private string _romType;
+        private Encoding _characterEncoding;
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="LcdConsole"/> class using the specified LCD low-level interface.
+        /// This class automatically configures the low-level interface. Do not use the low-level interface at the same time.
+        /// </summary>
+        /// <param name="lcd">The low-level LCD interface.</param>
+        /// <param name="romType">Name of character ROM of display. Currently supported types: A00 and A02.</param>
+        /// <param name="shouldDispose">If the class should dispose the LCD driver when it is disposed. Defaults to true</param>
+        public LcdConsole(ICharacterLcd lcd, string romType, bool shouldDispose = true)
```

Comment:
Correct me if I'm wrong, but seems like you should still be able to use the lcd1602 object to perform most of the operations r ight? so this just seems to be a wrapper on top of the Lcds. If so, I wonder if we should be forcing people to just use LcdConsole directly and not allow people to instantiate the low-level lcd interfaces.

Reply:
It's correct that you should not be using the underlying lcd1602 class at the same time as the LcdConsole class, because that would get the internal state of the LcdConsole class out of sync with the hardware. 
However, there are cases where you want to use the underlying class explicitly, i.e. if you want to do some more graphical approaches, or use it's hardware character scrolling capabilities. 
I've got another extension ready (https://github.com/pgrawehr/iot/blob/LCDValueUnitDisplay/src/devices/CharacterLcd/LcdValueUnitDisplay.cs - I'll prepare the PR once this one is merged) that uses block graphics to draw big numbers on a 20x4 display. This would be used instead of the LcdConsole class as alternate high-level api. 
So I think that the low-level interface is still required, even though you would normally just create it to feed it to (one of) the high-level apis. In the case I'm testing now, I am also able to use both those High-Level APIs interchangeably against the same low-level instance (display measurements in big, switch to normal font for menus)

Reply:
I see, that's ok, I guess what I meant my concern was is that we usually want to avoid having two ways of doing the same thing when possible, as that might confuse consumers which one they should use, but I guess most people trying to use this API will look into samples to check different usages so that will be fine. With that cleared up then this PR is mostly good to go, I think we are just waiting on this https://github.com/dotnet/iot/pull/859#discussion_r382423975 to be addressed as well as finalize what to do about https://github.com/dotnet/iot/pull/859#discussion_r391899360